unitTests/ui/QueryForCommerceTest.ts

```diff
@@ -0,0 +1,82 @@
+import { QueryForCommerce, IQueryForCommerceOptions } from '../../src/ui/QueryForCommerce/QueryForCommerce';
+import * as Mock from '../MockEnvironment';
+import { Simulate } from '../Simulate';
+import { AnalyticsEvents, $$ } from '../../src/Core';
+import { IChangeableAnalyticsDataObject, IChangeAnalyticsCustomDataEventArgs } from '../../src/events/AnalyticsEvents';
+import { analyticsActionCauseList } from '../../src/ui/Analytics/AnalyticsActionListMeta';
+
+export function QueryForCommerceTest() {
+  describe('QueryForCommerce', function() {
+    let test: Mock.IBasicComponentSetup<QueryForCommerce>;
+
+    const originalAnalyticsDataObject: IChangeableAnalyticsDataObject = {
+      language: 'en',
+      originLevel1: 'default',
+      originLevel2: 'All',
+      originLevel3: 'localhost:8080',
+      metaObject: {}
+    };
+
+    describe('with a listing value', () => {
+      beforeEach(function() {
+        test = Mock.optionsComponentSetup<QueryForCommerce, IQueryForCommerceOptions>(QueryForCommerce, <IQueryForCommerceOptions>{
+          listing: 'foobar'
+        });
+      });
+
+      it('should update tab in query builder to the listing option', function() {
+        const simulation = Simulate.query(test.env);
+        expect(simulation.queryBuilder.build().tab).toBe(test.cmp.options.listing);
+      });
+
+      it('should add a context value with the listing option', function() {
+        const simulation = Simulate.query(test.env);
+
+        expect(simulation.queryBuilder.build().context).toEqual(
```

Comment:
I think just doing 
```
expect(simulation.queryBuilder.build().context['listing']).toBe(test.cmp.options.listing)
```

is easier to read.

`jasmine.objectContaining` is helpful when you need to partially verify an object with multiple properties at once. In this case since we only wanna check one prop, it's a bit overkill.

Reply:
Cool